bolt-cli/src/common/mod.rs

```diff
@@ -105,3 +106,25 @@ pub fn try_parse_contract_error<T: SolInterface>(error: ContractError) -> Result
         _ => Err(error),
     }
 }
+
+/// Asks whether the user wants to proceed further. If not, the process is exited.
+#[allow(unreachable_code)]
+pub fn request_confirmation() {
+    // Skip confirmation in tests
+    #[cfg(test)]
+    return;
+
+    inquire::Confirm::new("Do you want to continue? (yes/no):")
+        .prompt()
+        .map(|proceed| {
+            if proceed {
+                return;
+            }
+            info!("Aborting");
+            std::process::exit(0);
+        })
+        .unwrap_or_else(|err| {
+            info!("confirmation exited: {}", err);
+            std::process::exit(0);
+        })
```

Comment:
Can you do some pattern matching based on `err`? Reference: https://docs.rs/inquire/latest/inquire/error/enum.InquireError.html

For example upon `OperationCanceled` or `OperatorInterrupted` we should log `"Aborting"` as well, otherwise log with `error` directive and return different error codes